pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1444,41 +1444,40 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
-			// completed, either successfully or with a failure.
-			if finalErr == nil {
-				// This can fail, but only for reasons that are okay (concurrent delete or update).
-				// Shouldn't happen in this case.
-				if isExtendedResourceClaim {
-					// Unlike other claims, extended resource claim is created in API server below.
-					// AssumeClaimAfterAPICall returns ErrNotFound when the informer update has not reached assumed cache yet.
-					// Hence we must poll and wait for it.
-					pollErr := wait.PollUntilContextTimeout(ctx, 1*time.Second, time.Duration(AssumeExtendedResourceTimeoutDefaultSeconds)*time.Second, true,
-						func(ctx context.Context) (bool, error) {
-							if err := pl.draManager.ResourceClaims().AssumeClaimAfterAPICall(claim); err != nil {
-								if errors.Is(err, assumecache.ErrNotFound) {
-									return false, nil
-								}
-								logger.V(5).Info("Claim not stored in assume cache", "claim", klog.KObj(claim), "err", err)
-								return false, err
+		// The scheduler was handling allocation. Now that has
+		// completed, either successfully or with a failure.
+		if resourceClaimModified {
+			// This can fail, but only for reasons that are okay (concurrent delete or update).
+			// Shouldn't happen in this case.
+			if isExtendedResourceClaim {
+				// Unlike other claims, extended resource claim is created in API server below.
+				// AssumeClaimAfterAPICall returns ErrNotFound when the informer update has not reached assumed cache yet.
+				// Hence we must poll and wait for it.
+				pollErr := wait.PollUntilContextTimeout(ctx, 1*time.Second, time.Duration(AssumeExtendedResourceTimeoutDefaultSeconds)*time.Second, true,
+					func(ctx context.Context) (bool, error) {
+						if err := pl.draManager.ResourceClaims().AssumeClaimAfterAPICall(claim); err != nil {
+							if errors.Is(err, assumecache.ErrNotFound) {
+								return false, nil
 							}
-							return true, nil
-						})
-					if pollErr != nil {
-						logger.V(5).Info("Claim not stored in assume cache after retries", "claim", klog.KObj(claim), "err", pollErr)
-					}
-				} else {
-					if err := pl.draManager.ResourceClaims().AssumeClaimAfterAPICall(claim); err != nil {
-						logger.V(5).Info("Claim not stored in assume cache", "err", err)
-					}
+							logger.V(5).Info("Claim not stored in assume cache", "claim", klog.KObj(claim), "err", err)
+							return false, err
+						}
+						return true, nil
+					})
+				if pollErr != nil {
+					logger.V(5).Info("Claim not stored in assume cache after retries", "claim", klog.KObj(claim), "err", pollErr)
+				}
+			} else {
+				if err := pl.draManager.ResourceClaims().AssumeClaimAfterAPICall(claim); err != nil {
+					logger.V(5).Info("Claim not stored in assume cache", "err", err)
 				}
 			}
+		}
 
-			for _, claimUID := range claimUIDs {
-				pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(claimUID)
-			}
+		for _, claimUID := range claimUIDs {
```

Comment:
This needs to remain under the `if allocation != nil` check.

Reply:
ok, added the check if allocation != nil